breakfastbot.py

```diff
@@ -60,7 +60,7 @@ def finish_poll():
             updater.bot.stop_poll(chat_id=poll[0], message_id=poll[1])
 
             # count the number of users who want bread
-            participant_count = len([x for option_ids in poll[2].values() if 0 in option_ids])
+            participant_count = len(["bread" for option_ids in poll[2].values() if 0 in option_ids])
```

Comment:
This change looks to me like it has no effect. What's this supposed to do?

Reply:
Fixing a "NameError: name 'x' is not defined"

Reply:
Can you please change it to
```Python
len([_ for option_ids in poll[2].values() if 0 in option_ids])
```
Using `_` is the idiomatic way to indicate that a variable/result is not used. Otherwise people will start to wonder what  `"bread"` is supposed to do.

Reply:
While testing on my console `_` is undefined before anything has been printed. Afterwards it just contains whatever the last output was. I dont really understand what the `_` in python actually does.

Therefore,
`len([_ for option_ids in poll[2].values() if 0 in option_ids])`
sometimes works and sometimes doesnt.
Using `"_"` or `""` would be options.

Reply:
I got confused here. The first entity of the list comprehension is not a declaration but an expression, so `_` cannot be used there. In that case I'd just suggest using `[option_ids for option_ids ...` to avoid introducing another name that might create confusion.